lemmy_structs/src/report.rs

```diff
@@ -0,0 +1,57 @@
+use lemmy_db::{comment_report::CommentReportView, post_report::PostReportView};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateReport {
+    pub report_type: String,
+    pub entity_id: i32,
+    pub reason: String,
+    pub auth: String,
+}
```

Comment:
This should be split out into CreateCommentReport, and CreatePostReport, instead of using report_type and entity_id

Then the api should be `/comment/report`, `/post/report`

Reply:
I suggested merging it into one API endpoint, because otherwise it was a lot of boilerplate code.